scim2.yaml

```diff
@@ -123,50 +156,63 @@ paths:
           type: string
         - in: body
           name: body
-          required: false
+          required: true
           schema:
-            type: string
+            $ref: '#/definitions/UserUpdateObject'
       responses:
-        '200':
+        200:
           description: User is updated
-        '401':
+          schema:
+            $ref: '#/definitions/UserResponseObject'
+        401:
           description: Unauthorized
-        '404':
+          schema:
+            $ref: '#/definitions/ErrorUnauthorized'
+        404:
           description: Valid user is not found
+          schema:
+            $ref: '#/definitions/ErrorUserNotAvailable'
     delete:
       tags:
         - Me Endpoint
       summary: Delete Me
-      description: This API is used to delete the currently authenticated user. It returns HTTP 204 if the user is successfully deleted.
+      description: |
+        This API is used to delete the currently authenticated user. It returns HTTP 204 if the user is successfully deleted.
+
+          <b>Permission required:</b>
+
+            * /permission/admin/manage/identity/usermgt/delete
       operationId: deleteUserMe
-      x-wso2-request: |
-         curl -v -k --user  [username]:[password] -X DELETE https://localhost:9443/scim2/Me
-      x-wso2-curl: |
-         curl -v -k --user kim:kimwso2 -X DELETE https://localhost:9443/scim2/Me
-      x-wso2-response: |
-        HTTP/1.1 501 Not Implemented
       produces:
         - application/scim+json
       parameters: []
       responses:
-        '204':
+        204:
           description: User is deleted
-        '401':
+        401:
           description: Unauthorized
-        '404':
+          schema:
+            $ref: '#/definitions/ErrorUnauthorized'
+        404:
           description: Valid user is not found
+          schema:
+            $ref: '#/definitions/ErrorUserNotAvailable'
+        501:
+          description: Self delete is not suported
+          schema:
+            $ref: '#/definitions/ErrorNotimplemented'
     patch:
       tags:
         - Me Endpoint
-      summary: Update Me - PATCH 
-      description: This API uses a PATCH operation to update user details. It returns HTTP 404 if the user is not found.
+      summary: Update Me - PATCH
+      description: |
+        This API uses a PATCH operation to update user details. It returns HTTP 404 if the user is not found.
```

Comment:
Since the operation and the response codes are defined we can omit these from the description and describe what happens when the API is invoked